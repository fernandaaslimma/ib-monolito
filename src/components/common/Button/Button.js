import React from "react";
import { bool, node, string, func, number } from "prop-types";
import {
  ButtonTag,
  Content,
  WrapperChildren,
  WrapperIcon
} from "../../../styles/objects";
import LocalLoading from "../LocalLoading/";
import Icon from "../Icon";

function Button({
  children,
  onClick,
  isCallToAction,
  isWarning,
  type,
  disabled,
  width,
  className,
  small,
  height,
  dataTest,
  paddingSize,
  opacity,
  approveButton,
  actionSecondary,
  loading,
  background,
  color,
  border,
  fontSize,
  borderRadius,
  noBorderRadius,
  margin,
  padding,
  noHoverBackground,
  smallBlue,
  backgroundLoading,
  id,
  withIcon,
  isModalType,
  style
}) {
  return (
    <ButtonTag
      id={id}
      className={className}
      width={width}
      height={height}
      approveButton={approveButton}
      type={type}
      disabled={disabled}
      isCallToAction={isCallToAction}
      isWarning={isWarning}
      data-test={dataTest ? dataTest : Button.displayName}
      onClick={onClick}
      small={small}
      smallBlue={smallBlue}
      opacity={opacity}
      paddingSize={paddingSize}
      actionSecondary={actionSecondary}
      loading={loading}
      background={background}
      color={color}
      border={border}
      fontSize={fontSize}
      borderRadius={borderRadius}
      noBorderRadius={noBorderRadius}
      margin={margin}
      padding={padding}
      noHoverBackground={noHoverBackground}
      isModalType={isModalType}
      style={style}
    >
      <Content>
        {withIcon && !loading && (
          <WrapperIcon>
            <Icon
              type={withIcon.name}
              width={withIcon.width}
              height={withIcon.height}
            ></Icon>
          </WrapperIcon>
        )}
        {loading ? (
          <LocalLoading background={backgroundLoading} />
        ) : (
          <WrapperChildren withIcon={withIcon}>{children}</WrapperChildren>
        )}
      </Content>
    </ButtonTag>
  );
}

Button.displayName = "Button";

Button.defaultProps = {
  isCallToAction: null,
  isWarning: null,
  type: "button",
  disabled: false,
  width: null,
  height: null,
  onClick: null,
  className: null,
  small: false,
  paddingSize: null,
  actionSecondary: null
};

Button.propTypes = {
  children: node,
  isCallToAction: bool,
  isWarning: bool,
  disabled: bool,
  type: string,
  width: number,
  onClick: func,
  className: string,
  small: bool,
  height: number,
  paddingSize: number,
  actionSecondary: bool
};

export default Button;
