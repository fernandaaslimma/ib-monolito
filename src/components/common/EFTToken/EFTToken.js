import React, { Component } from "react";
import { func } from "prop-types";
import Icon from "../Icon";
import Input from "../Input";
import { translate } from "../../../utils/i18n";
import { starifyMail, starifyTelephone } from "../../../utils/starify";
import { telephoneMask } from "../../../utils/masks";
import { conformToMask } from "react-text-mask";
import SelectMethod from "./SelectMethod";
import { Button } from "react-bocombbm-components";
import {
  EFTModalWrapper,
  IconWrapper,
  Title,
  Message,
  ErrorMessage,
  FooterContainer,
  Padding,
  Disclaimer,
  Identificator,
  LinksWrapper,
  Loader,
  MfaWrapper,
  ButtonWrapper
} from "./styles";
import Timer from "../Timer";
import ClickWrapper from "../../../utils/clickWrapper";

class EFTToken extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isLoading: false,
      contentLoaded: false,
      validInput: false,
      relativeFactors: [],
      authFactor: [],
      showTimeOut: false
    };

    this.resetTimeOutStates = this.resetTimeOutStates.bind(this);
    this.sendAnotherToken = this.sendAnotherToken.bind(this);
    this.handleConfirmButton = this.handleConfirmButton.bind(this);
    this.changeCurrentMethod = this.changeCurrentMethod.bind(this);
    this.ref = React.createRef();
  }

  selectMFAMethod() {
    //this.setState({ methodChanging: true });
    this.props.changeFactorTogle(true);
  }

  async changeCurrentMethod(receivedMethodId) {
    await this.recreateAuthFactorCode(receivedMethodId);
    const receivedMethod = this.state.relativeFactors.filter(
      method => method.id === receivedMethodId
    );
    this.setState({ authFactor: receivedMethod[0] });
    this.props.changeFactorTogle(false);
  }

  sendAnotherToken() {
    this.setState({ showTimeOut: true });

    const { openToastr } = this.props;
    openToastr({
      text: translate("RESEND_TOKEN_EMAIL"),
      isBelow: false,
      isTop: true
    });
    this.recreateAuthFactorCode(this.state.authFactor.id);
  }

  async recreateAuthFactorCode(authFactorId) {
    const {
      createAuthCode,
      createAuthCodeParams,
      handleMFAInputClean,
      saveAccount
    } = this.props;
    const { payload, actionType } = createAuthCodeParams;

    await createAuthCode(payload, actionType, authFactorId, saveAccount);
    handleMFAInputClean();
  }

  async createAuthFactorCode() {
    const {
      getAuthFactors,
      createAuthCode,
      createAuthCodeParams,
      saveAccount
    } = this.props;
    const { payload, actionType } = createAuthCodeParams;

    await getAuthFactors();

    const relativeFactors = await this.props.authFactors.filter(factor => {
      if (factor.type === "mobile") {
        return false;
      }
      if (factor.actions.includes(actionType)) {
        return true;
      }
    });

    const authFactor = relativeFactors.reduce((acc, auth) => {
      if (auth.defaultAuth === true) {
        return auth;
      }
      return acc;
    });

    await createAuthCode(payload, actionType, authFactor.id, saveAccount);

    if (this.props.authFactors.length > 0) {
      this.setState({
        contentLoaded: true,
        authFactor: authFactor,
        relativeFactors: relativeFactors
      });
    }
  }

  async componentDidMount() {
    await this.props.changeFactorTogle(false);
    await this.createAuthFactorCode();
  }

  componentWillUnmount() {
    this.props.clearMFAToken();
    this.props.clearMFATokenValidated();
  }

  resetTimeOutStates() {
    this.setState({
      showTimeOut: false
    });
  }

  handleInputChange(...e) {
    this.setState({
      validInput: false
    });
    this.props.handleMFAInputToken(...e);
  }

  handleConfirmButton() {
    this.setState({ isLoading: true });

    this.props
      .checkMFA()
      .then(data => {
        this.props.onMFAConfirmation(data);
      })
      .catch(err => {
        this.setState({
          isLoading: false,
          validInput: true
        });
        this.props.onMFAError(err);
      });
  }

  getMFAContent() {
    const { type, authUri } = this.state.authFactor;
    const dict = {
      mail: {
        icon: "Email",
        disclaimer: ` ${translate("MFA_EMAIL_DISCLAIMER")}`,
        indentificator: authUri ? starifyMail(authUri) : null,
        disclaimer2: ` ${translate("MFA_EMAIL_DISCLAIMER_2")}`,
        iconWidth: "96",
        iconHeight: "96",
        type
      },
      sms: {
        icon: "Sms",
        disclaimer: translate("MFA_SMS_DISCLAIMER"),
        indentificator: authUri
          ? conformToMask(starifyTelephone(authUri), telephoneMask)
              .conformedValue
          : null,
        type
      },
      mobile: {
        icon: "Smartphone",
        disclaimer: translate("MFA_APP_DISCLAIMER_1"),
        indentificator: translate("MFA_APP_DISCLAIMER_APP"),
        disclaimer2: translate("MFA_APP_DISCLAIMER_3"),
        iconWidth: "56",
        iconHeight: "100",
        type
      },
      totp: {
        icon: "MfaAuthenticator",
        disclaimer: translate("MFA_TOTP_DISCLAIMER"),
        iconWidth: "96",
        iconHeight: "96",
        type
      }
    };

    return dict[type];
  }

  renderContent() {
    const { mfaTokenValidated, mfaToken } = this.props;
    const { isLoading, contentLoaded, showTimeOut } = this.state;
    const enableMFAChange = this.state.relativeFactors.length > 1;

    if (!contentLoaded) {
      return <Loader />;
    }

    const {
      icon,
      disclaimer,
      disclaimer2,
      indentificator,
      iconWidth = "89",
      iconHeight = "58",
      type
    } = this.getMFAContent();

    return (
      <MfaWrapper>
        <IconWrapper>
          <Icon width={iconWidth} height={iconHeight} type={icon} />
        </IconWrapper>
        <Padding>
          <Disclaimer data-test="methodDisclaimer">
            {disclaimer}
            {indentificator && <Identificator>{indentificator}</Identificator>}
            {disclaimer2 && <span> {disclaimer2}</span>}
          </Disclaimer>
        </Padding>

        <ButtonWrapper>
          {type !== "mobile" && (
            <Input
              icon="Lock"
              onChange={e => this.handleInputChange(e)}
              type="password"
              name="token"
              valid={() => mfaTokenValidated}
              value={mfaToken}
              label={translate("TOKEN")}
              tinyLabels
              disableKeys={[13]}
              dataTest="TransactionTokenField"
              disabled={isLoading}
              innerRef={this.ref}
              inputMode={
                window.matchMedia("(max-width: 760px)").matches
                  ? "numeric"
                  : "text"
              }
            />
          )}

          <ErrorMessage
            data-test={`ModalFeedbackMsg${mfaTokenValidated === false ||
              this.state.validInput}`}
            visible={mfaTokenValidated === false || this.state.validInput}
          >
            {translate("TOKEN_ERROR_MESSAGE")}
          </ErrorMessage>
          {type !== "mobile" && (
            <ClickWrapper>
              <Button
                spacing={{ top: "xs", bottom: "s" }}
                id="Confirm"
                type="conclusive"
                block
                onClick={() => this.handleConfirmButton()}
                disabled={mfaToken.length < 1 || !contentLoaded}
                loading={!!isLoading}
                dataTest="Confirm"
              >
                {translate("MFA_CONFIRM_BTN")}
              </Button>
            </ClickWrapper>
          )}
        </ButtonWrapper>
        <FooterContainer>
          <LinksWrapper>
            {enableMFAChange && (
              <ClickWrapper>
                <Button
                  type="text"
                  spacing={{
                    top: "xs",
                    bottom: "none",
                    right: "none",
                    left: "none"
                  }}
                  small
                  dataTest="selectMFAMethod"
                  onClick={() => this.selectMFAMethod()}
                >
                  {translate("CHANGE_MFA_METHOD_START_BUTTON")}
                </Button>
              </ClickWrapper>
            )}

            {type === "mail" ? (
              showTimeOut ? (
                <ClickWrapper>
                  <Timer
                    startTimer={showTimeOut}
                    timeLeft={30}
                    expirationAction={() => this.resetTimeOutStates()}
                  />
                </ClickWrapper>
              ) : (
                <ClickWrapper>
                  <Button
                    type="text"
                    spacing={{ top: "xs", bottom: "none" }}
                    small
                    dataTest="ResendToken"
                    onClick={() => this.sendAnotherToken()}
                  >
                    {translate("MFA_RESEND_TOKEN")}
                  </Button>
                </ClickWrapper>
              )
            ) : null}
          </LinksWrapper>
        </FooterContainer>
      </MfaWrapper>
    );
  }

  render() {
    const { type } = this.state.authFactor;
    const methodsToSelect = this.state.relativeFactors.filter(
      method => method.type != type
    );
    const { title, noMessage, message } = this.props;
    return !this.props.methodChanging ? (
      <EFTModalWrapper>
        <div>
          {title ? (
            <Title>{title}</Title>
          ) : (
            <Title>{translate("MFA_TITLE")}</Title>
          )}
          {noMessage ? null : message ? (
            <Message>{message}</Message>
          ) : (
            <Message>{translate("MFA_MESSAGE")}</Message>
          )}

          {this.renderContent()}
        </div>
      </EFTModalWrapper>
    ) : (
      <EFTModalWrapper>
        <div>
          <Title>{translate("CHANGE_MFA_METHOD_TITLE")}</Title>
          {noMessage ? null : message ? (
            <Message>{message}</Message>
          ) : (
            <Message>{translate("CHANGE_MFA_METHOD_MESSAGE")}</Message>
          )}

          <MfaWrapper>
            <SelectMethod
              methods={methodsToSelect}
              changeCurrentMethod={this.changeCurrentMethod}
            />
          </MfaWrapper>
        </div>
      </EFTModalWrapper>
    );
  }
}

EFTToken.defaultProps = {
  mfaToken: "",
  //Default onMFAError function in case of not be sended
  //Best point to implement global MFA error treathment
  onMFAError: () => {}
};

EFTToken.propTypes = {
  onMFAConfirmation: func.isRequired,
  onMFAError: func.isRequired
};

export default EFTToken;
