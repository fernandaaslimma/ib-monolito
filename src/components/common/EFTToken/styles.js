import styled, { css } from "styled-components";
import { rem, remFontSize, media } from "../../../styles/tools";
import { blue70, lightRed, darkBlue, gray200 } from "../../../styles/settings";
import { InputWrapper } from "../../common/Input/styles";
import { LinkTag } from "../../../styles/objects/actionElements";
import LocalLoading from "../LocalLoading";
import { Bar } from "../LocalLoading/styles";

export const EFTModalWrapper = styled.div`
  width: calc(100% - ${rem(32)});
  ${InputWrapper} {
    margin-top: ${rem(20)};
    height: ${rem(48)};
  }

  ${LinkTag} {
    margin-top: ${rem(20)};
    height: ${rem(40)};
    line-height: ${rem(40)};
    width: ${rem(120)};
  }
  margin: auto ${rem(16)} auto ${rem(16)};

  ${media.md(css`
    width: ${rem(382)};
    margin: auto;
  `)};
`;

export const Loader = styled(LocalLoading)`
  height: ${rem(175)};
  position: relative;
  ${Bar} {
    background-color: ${blue70};
  }
`;
export const IconWrapper = styled.div`
  width: 100%;
  margin: 0 auto;
  margin: ${rem(20)} 0 ${rem(12)} 0;
`;

export const Title = styled.h1`
  font-family: Lato Bold;
  font-weight: bolder;
  font-size: ${remFontSize(20)};
  text-align: center;
  letter-spacing: ${rem(0.49)};
  margin: ${rem(23)} ${rem(0)} ${rem(0)};
  color: ${blue70};

  ${media.md(css`
    margin: ${rem(40)} auto ${rem(10)} auto;
    font-size: ${remFontSize(24)};
  `)};
`;

export const Message = styled.div`
  font-size: ${remFontSize(16)};
  font-family: Lato;
  margin: ${rem(12)} auto 0 auto;
  color: ${gray200};
  line-height: ${remFontSize(22)};

  ${media.md(css`
    letter-spacing: ${rem(0.42)};
    font-size: ${remFontSize(18)};
    max-width: ${rem(382)};
    margin: ${rem(12)} auto 0 auto;
  `)};
`;

export const Disclaimer = styled(Message)`
  letter-spacing: ${rem(0.39)};
  width: 100%;
  margin: ${rem(0)};
  font-size: ${remFontSize(14)};
  ${media.md(css`
    font-size: ${remFontSize(13)};
    width: ${rem(345)};
  `)};
  vertical-align: baseline;
`;

export const Identificator = styled.span`
  max-width: 100%;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  font-family: Lato Bold;
  color: ${blue70};
  vertical-align: baseline;
`;

export const Padding = styled.div`
  display: flex;
  width: 100%;
  flex-direction: column;
  align-items: center;
  padding: 0 0;
`;

export const ErrorMessage = styled.h1`
  display: none;

  ${({ visible }) =>
    visible &&
    css`
      display: block;
      font-size: ${remFontSize(13)};
      font-style: italic;
      font-family: Lato;
      text-align: center;
      margin: 0 auto;
      margin: ${rem(5)};
      color: ${lightRed};
    `};
`;

export const Link = styled.a`
  font-family: Lato;
  font-size: ${remFontSize(13)};
  text-decoration: none;
  text-align: center;
  letter-spacing: ${rem(0.42)};
  color: ${darkBlue};
  :focus,
  :hover {
    text-decoration: none;
  }
  :only-child {
    margin-right: 0;
  }
`;

export const LinksWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;

  ${media.md(css`
    margin: auto;
    margin-bottom: ${rem(10)};
    max-width: ${rem(350)};
  `)};
`;

export const FooterContainer = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const MfaWrapper = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: center;

  ${media.md(css`
    margin: auto;
  `)};
`;

export const ButtonWrapper = styled.div`
  ${media.md(css`
    margin: auto;
    width: ${rem(275)};
  `)};
`;
