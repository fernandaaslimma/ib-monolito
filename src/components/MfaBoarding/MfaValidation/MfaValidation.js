import React, { Fragment } from "react";
import EFTToken from "../../common/EFTToken";
import { translate } from "../../../utils/i18n";
import { Button } from "react-bocombbm-components";

import ClickWrapper from "../../../utils/clickWrapper/clickWrapper";

import { ACTION_TYPE_APPOVE_TOTP_FACTOR } from "../../../utils/constants";

function MfaValidation({
  changeCurrentScreen,
  authFactorResponse,
  isMfaExchange
}) {
  const { authFactorID } = authFactorResponse;
  const params = {
    actionType: ACTION_TYPE_APPOVE_TOTP_FACTOR,
    payload: { id: authFactorID }
  };

  return (
    <Fragment>
      <EFTToken
        resendFunc={e => e.preventDefault()}
        title={translate("MFA_BOARDING_TITLE")}
        onMFAConfirmation={() =>
          isMfaExchange ? changeCurrentScreen(3) : changeCurrentScreen(6)
        }
        createAuthCodeParams={params}
        onMFAError={() => {}}
      />

      <ClickWrapper>
        <Button
          spacing={{ bottom: "m", top: "xs" }}
          type="text"
          small
          dataTest="RegisterLater"
          onClick={() => changeCurrentScreen(2)}
        >
          {translate("REGISTER_LATER")}
        </Button>
      </ClickWrapper>
    </Fragment>
  );
}

export default MfaValidation;
