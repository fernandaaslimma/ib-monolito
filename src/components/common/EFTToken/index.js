import EFTToken from "./EFTToken";

import { connect } from "redux-zero/react";
import { combineActions } from "redux-zero/utils";
import MFATokenActions from "../../../actions/mfa";
import toastrActions from "../../../actions/toastr";

export default connect(
  ({ mfaToken, mfaTokenValidated, authFactors, methodChanging, ...props }) => ({
    mfaToken,
    mfaTokenValidated,
    authFactors,
    methodChanging,
    props
  }),
  combineActions(MFATokenActions, toastrActions)
)(EFTToken);
