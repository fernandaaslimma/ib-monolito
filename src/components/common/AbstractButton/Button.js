import React from "react";
import { bool, node, string, func } from "prop-types";
import { ButtonTag } from "./styles";

function Button({
  children,
  onClick,
  isCallToAction,
  isWarning,
  isSecondary,
  isSpanButton,
  type,
  disabled,
  width,
  className,
  small,
  dataTest,
  opacity,
  height,
  noPadding,
  heightPx
}) {
  return (
    <ButtonTag
      className={className}
      width={width}
      type={type}
      disabled={disabled}
      isCallToAction={isCallToAction}
      isWarning={isWarning}
      isSecondary={isSecondary}
      isSpanButton={isSpanButton}
      data-test={dataTest ? dataTest : Button.displayName}
      onClick={onClick}
      small={small}
      opacity={opacity}
      height={height}
      noPadding={noPadding}
      heightPx={heightPx}
    >
      {children}
    </ButtonTag>
  );
}

Button.displayName = "Button";

Button.defaultProps = {
  isCallToAction: null,
  isWarning: null,
  type: "button",
  disabled: false,
  width: null,
  onClick: null,
  className: null,
  small: false
};

Button.propTypes = {
  children: node.isRequired, // eslint-disable-line
  isCallToAction: bool,
  isWarning: bool,
  isSecondary: bool,
  isSpanButton: bool,
  disabled: bool,
  type: string,
  width: string,
  onClick: func,
  className: string,
  small: bool,
  dataTest: string,
  opacity: string,
  height: string,
  noPadding: bool,
  heightPx: string
};

export default Button;
