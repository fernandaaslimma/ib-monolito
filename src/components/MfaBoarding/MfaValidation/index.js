import MfaValidation from "./MfaValidation";

export default MfaValidation;
