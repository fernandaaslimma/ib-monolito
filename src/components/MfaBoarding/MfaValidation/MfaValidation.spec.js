import React from "react";
import { shallow } from "enzyme";
import MfaValidation from "./MfaValidation";
import EFTToken from "../../common/EFTToken";
import Button from "react-bocombbm-components/dist/Button";
const props = {
  props: { mfaTokenValidated: undefined },
  changeCurrentScreen: jest.fn(),
  preventDefault: jest.fn(),
  authFactorResponse: {
    authFactorID: "29c6368f-cc9b-4458-8db9-b56e62b2f4d4",
    activationURL:
      "otpauth://totp/BOCOM%20BBM:pj1@bocombbm.com.br?secret=XOSONHXJDCDGSC2YX5TIXL3OIE======&issuer=BOCOM%20BBM"
  }
};

describe("MfaValidation component", () => {
  it("should match snapshot with props", () => {
    expect(shallow(<MfaValidation {...props} />)).toMatchSnapshot();
  });

  it("should call changeCurrentPage onMFAConfirmation", () => {
    const shallowComponent = shallow(<MfaValidation {...props} />);
    shallowComponent.find(EFTToken).prop("onMFAConfirmation")();
    expect(props.changeCurrentScreen).toHaveBeenCalledWith(6);
  });

  it("should call changeCurrentPage onMFAConfirmation", () => {
    const shallowComponent = shallow(<MfaValidation {...props} />);
    shallowComponent.find(EFTToken).prop("onMFAError")();
    expect(shallowComponent).toMatchSnapshot();
  });

  it("should call changeCurrentScreen on click Button", () => {
    const shallowComponent = shallow(<MfaValidation {...props} />);
    shallowComponent.find(Button).simulate("click");
    expect(props.changeCurrentScreen).toHaveBeenCalledWith(2);
  });
});
