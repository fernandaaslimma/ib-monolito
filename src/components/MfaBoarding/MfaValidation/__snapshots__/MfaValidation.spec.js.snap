// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MfaValidation component should call changeCurrentPage onMFAConfirmation 1`] = `
<Fragment>
  <ConnectWrapper
    createAuthCodeParams={
      Object {
        "actionType": "approvetotpfactor",
        "payload": Object {
          "id": "29c6368f-cc9b-4458-8db9-b56e62b2f4d4",
        },
      }
    }
    onMFAConfirmation={[Function]}
    onMFAError={[Function]}
    resendFunc={[Function]}
  />
  <ClickWrapper>
    <Button
      dataTest="RegisterLater"
      disabled={false}
      onClick={[Function]}
      small={true}
      spacing={
        Object {
          "bottom": "m",
          "top": "xs",
        }
      }
      type="text"
    />
  </ClickWrapper>
</Fragment>
`;

exports[`MfaValidation component should match snapshot with props 1`] = `
<Fragment>
  <ConnectWrapper
    createAuthCodeParams={
      Object {
        "actionType": "approvetotpfactor",
        "payload": Object {
          "id": "29c6368f-cc9b-4458-8db9-b56e62b2f4d4",
        },
      }
    }
    onMFAConfirmation={[Function]}
    onMFAError={[Function]}
    resendFunc={[Function]}
  />
  <ClickWrapper>
    <Button
      dataTest="RegisterLater"
      disabled={false}
      onClick={[Function]}
      small={true}
      spacing={
        Object {
          "bottom": "m",
          "top": "xs",
        }
      }
      type="text"
    />
  </ClickWrapper>
</Fragment>
`;
